### ISSUE TITLE
Parsing large files is slow

### ISSUE THREAD

**Issue by jacebrowning** [sdispater/tomlkit#10]
Parsing large TOML files is prohibitively slow. On a large Django application with 230 dependencies, Poetry has generated a 4145 line, 5 KB `pyproject.lock` file that takes more than 4 minutes to parse on my iMac:

```
$ time .venv/bin/python -c "import tomlkit; tomlkit.parse(open('pyproject.lock').read())"
250.63s user 2.00s system 99% cpu 4:13.40 total
```

If I break early, it always seems to be stuck in `_restore_idx`:

```
>>> tomlkit.parse(text)
^CTraceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../.venv/lib/python2.7/site-packages/tomlkit/api.py", line 51, in parse
    return Parser(string).parse()
  File ".../.venv/lib/python2.7/site-packages/tomlkit/parser.py", line 144, in parse
    key, value = self._parse_table()
  File ".../.venv/lib/python2.7/site-packages/tomlkit/parser.py", line 959, in _parse_table
    result = self._parse_aot(result, name)
  File ".../.venv/lib/python2.7/site-packages/tomlkit/parser.py", line 1008, in _parse_aot
    _, table = self._parse_table(name_first)
  File ".../.venv/lib/python2.7/site-packages/tomlkit/parser.py", line 927, in _parse_table
    key_next, table_next = self._parse_table(name)
  File ".../.venv/lib/python2.7/site-packages/tomlkit/parser.py", line 924, in _parse_table
    is_aot_next, name_next = self._peek_table()
  File ".../.venv/lib/python2.7/site-packages/tomlkit/parser.py", line 993, in _peek_table
    self._restore_idx(*idx)
  File ".../.venv/lib/python2.7/site-packages/tomlkit/parser.py", line 301, in _restore_idx
    [(i + idx, TOMLChar(c)) for i, c in enumerate(self._src[idx:])]
  File ".../.venv/lib/python2.7/site-packages/tomlkit/toml_char.py", line 8, in __init__
    super(TOMLChar, self).__init__()
KeyboardInterrupt
```

**Comment by sdispater** [comment]
That's a lot of time!

Could you check if it's faster to parse the file on Python >= 3.4, since I had to implement a workaround for Python 2.7 in `_restore_idx()`.

**Comment by jacebrowning** [comment]
The same file takes 0.35 seconds to parse with Python 3.7. 😆

**Comment by sdispater** [comment]
This is fixed in the latest `0.4.4` release!

**Comment by jacebrowning** [comment]
Awesome! I'm looking forward to a bugfix release of Poetry that depends on this new version.

https://github.com/sdispater/poetry/pull/418
